code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -540,3 +544,207 @@
 	owner.remove_actionspeed_modifier(/datum/actionspeed_modifier/nooartrium)
+
+/*				Non c2 medicines 				*/
+
+/datum/reagent/impurity/mannitol
+	name = "Mannitoil"
+	description = "Gives the patient a temporary speech impediment."
+	color = "#CDCDFF"
+	addiction_types = list(/datum/addiction/medicine = 5)
+	ph = 12.4
+	liver_damage = 0
+	///The speech we're forcing on the owner
+	var/speech_option
+
+/datum/reagent/impurity/mannitol/on_mob_add(mob/living/owner, amount)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon = owner
+	speech_option = pick(list(SWEDISH, UNINTELLIGIBLE, STONER, MEDIEVAL, WACKY, NERVOUS, MUT_MUTE))
+	carbon.dna?.activate_mutation(speech_option)
+
+/datum/reagent/impurity/mannitol/on_mob_delete(mob/living/owner)
+	. = ..()
+	if(!iscarbon(owner))
+		return
+	var/mob/living/carbon/carbon = owner
+	carbon.dna?.remove_mutation(speech_option)
+
+/datum/reagent/inverse/neurine
+	name = "Neruwhine"
+	description = "Induces a temporary brain trauma in the patient by redirecting neuron activity."
+	color = "#DCDCAA"
+	ph = 13.4
+	addiction_types = list(/datum/addiction/medicine = 8)
+	metabolization_rate = 0.025 * REM
+	tox_damage = 0
+	//The temporary trauma passed to the owner
+	var/datum/brain_trauma/temp_trauma
+
+/datum/reagent/inverse/neurine/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
+	.=..()
+	if(temp_trauma)
+		return
+	if(!(DT_PROB(creation_purity*10, delta_time)))
+		return
+	var/traumalist = subtypesof(/datum/brain_trauma)
+	traumalist -= /datum/brain_trauma/severe/split_personality //Uses a ghost, I don't want to use a ghost for a temp thing.
+	var/datum/brain_trauma/trauma = pick(traumalist)
+	var/obj/item/organ/brain/brain = owner.getorganslot(ORGAN_SLOT_BRAIN)
+	if(brain.brain_gain_trauma(trauma, TRAUMA_RESILIENCE_MAGIC))
+		temp_trauma = trauma
+
+/datum/reagent/inverse/neurine/on_mob_delete(mob/living/carbon/owner)
+	.=..()
+	if(!temp_trauma)
+		return
+	if(istype(temp_trauma, /datum/brain_trauma/special/imaginary_friend))//Good friends stay by you, no matter what
+		return
+	owner.cure_trauma_type(temp_trauma, resilience = TRAUMA_RESILIENCE_MAGIC)
+
+/datum/reagent/inverse/corazone
+	name = "Corazargh" //It's what you yell! Though, if you've a better name feel free. Also an omage to an older chem
+	description = "Interferes with the body's natural pacemaker, forcing the patient to manually beat their heart."
+	color = "#5F5F5F"
+	self_consuming = TRUE
+	ph = 13.5
+	addiction_types = list(/datum/addiction/medicine = 2.5)
+	metabolization_rate = 0.01 * REM
+	chemical_flags = REAGENT_DEAD_PROCESS
+	tox_damage = 0
+	///The old heart we're swapping for
+	var/obj/item/organ/heart/original_heart
+	///The new heart that's temp added
+	var/obj/item/organ/heart/cursed/manual_heart
+
+///Creates a new cursed heart and puts the old inside of it, then replaces the position of the old
+/datum/reagent/inverse/corazone/on_mob_add(mob/living/owner)
```

Comment:
this and the on_mob_delete() effect should probably use on_mob_metabolize and on_mob_end_metabolize() instead.

Reply:
Will do!